sdk/ml/azure-ai-ml/azure/ai/ml/operations/_code_operations.py

```diff
@@ -92,9 +100,36 @@ def create_or_update(self, code: Code) -> Code:
                     registry=self._registry_name,
                     body=get_asset_body_for_registry_storage(self._registry_name, "codes", name, version),
                 )
-            code, _ = _check_and_upload_path(
-                artifact=code, asset_operations=self, sas_uri=sas_uri, artifact_type=ErrorTarget.CODE
-            )
+                code, _ = _check_and_upload_path(
+                    artifact=code,
+                    asset_operations=self,
+                    sas_uri=sas_uri,
+                    artifact_type=ErrorTarget.CODE
+                )
+            else:
+                path, ignore_file, asset_hash = _get_snapshot_path_info(code)
+                workspace_info = self._datastore_operation._service_client.workspaces.get(
+                    resource_group_name=self._resource_group_name,
+                    workspace_name=self._workspace_name
+                )
+                existing_asset = _get_existing_snapshot_by_hash(
```

Comment:
Why don't we need to check the existence of registry assets?

Reply:
It's already checked with this call: `get_sas_uri_for_registry_asset`. 

It's calling the `create_or_get_temporary_data_reference` method from the service client. We can't use that here for snapshots because it requires a registry input. The Grace Bramley Simmons and Anthony Karloff are working to make sure the next MFE contract will be amended to have a non-registry option that can be used here: https://msdata.visualstudio.com/Vienna/_git/vienna/pullrequest/912910. But in the meantime, we have to implement the non-registry version with API calls in the client.